dan_layer/core/src/workers/pacemaker_worker.rs

```diff
@@ -104,34 +104,38 @@ where T: Clone + Debug + PartialEq + Eq + Hash + Send + Sync + 'static
                             self.handle_stop_signal(wait_over);
                         }
                         PacemakerTimer::Start(wait_over, duration_timeout) => {
-                    info!(
-                        target: LOG_TARGET,
-                        "Received start wait signal for value: {:?}", wait_over
-                    );
-                    let (tx, rx_stop_signal) = oneshot::channel();
-                    if self.pending_timeouts.insert(wait_over.clone(), tx).is_none() {
```

Comment:
Was this the source of the bug ? That you always inserted a new pending signal, independently if it was already there ?

Reply:
One of them. What this was doing is best described by the documentation `If the map did have this key present, the value is updated, and the old value is returned` from that you can imagine the implications.